Basic/Invaders/Assets/Scripts/InvadersGame.cs

```diff
@@ -78,31 +75,40 @@ public class InvadersGame : NetworkBehaviour
     private bool m_ClientGameStarted;
     private bool m_ClientStartCountdown;
 
-    private NetworkVariableBool m_CountdownStarted = new NetworkVariableBool(false);
+    private NetworkVariable<bool> m_CountdownStarted = new NetworkVariable<bool>(false);
 
     private float m_NextTick;
 
     // the timer should only be synced at the beginning
     // and then let the client to update it in a predictive manner
-    private NetworkVariableFloat m_ReplicatedTimeRemaining = new NetworkVariableFloat();
+    private bool m_ReplicatedTimeSent = false;
     private GameObject m_Saucer;
     private List<Shield> m_Shields = new List<Shield>();
     private float m_TimeRemaining;
 
     public static InvadersGame Singleton { get; private set; }
 
-    public NetworkVariableBool hasGameStarted { get; } = new NetworkVariableBool(false);
+    public NetworkVariable<bool> hasGameStarted { get; } = new NetworkVariable<bool>(false);
 
-    public NetworkVariableBool isGameOver { get; } = new NetworkVariableBool(false);
+    public NetworkVariable<bool> isGameOver { get; } = new NetworkVariable<bool>(false);
 
     /// <summary>
     ///     Awake
     ///     A good time to initialize server side values
     /// </summary>
     private void Awake()
     {
-        // TODO: Improve this singleton pattern
-        Singleton = this;
+        if (Singleton != null && Singleton != this)
```

Comment:
I like the Assert based idea more

Reply:
The problem I found with the Assert method is that does not allow to have the Singleton manager on each scene that may have it for debug purposes for example. With the Destroy pattern it allows you to have managers on all you Scenes and it will be just destroyed if you come from another Scene.

Reply:
You are totally right I was totally blinded by the Singleton name of the variable that meant for me for the duration of the game not only one Scene. Updated it.